File: plot/python/2D_BitmapMode_proto/main.py
from md import *


# ---------------------------------------------------------------------------------------
#                                        OPENGL DRAW
# ---------------------------------------------------------------------------------------
def draw():  # you can write some GL sequence in this fuction if needed
    draw_pre()
    loop()
    for pixel in VMD.pixels:
        draw_pixel(pixel.x, pixel.y)
    for line in VMD.lines:
        draw_line(line.x, line.y, line.w, line.z)
    draw_post()


# ---------------------------------------------------------------------------------------
#                                       END OPENGL DRAW
# ---------------------------------------------------------------------------------------


# /////////////////////////////////////////////////////////////
#                       MAIN
# /////////////////////////////////////////////////////////////

def main():
    palette_green = 'palette_green'

    SGDK.c('int main(){')
    SGDK.VDP_setScreenWidth256()
    SGDK.VDP_setPalette(0, palette_green)
    SGDK.BMP_init(TRUE, 0, FALSE)
    render()


def loop():
    global line001
    SGDK.c('while(1){')
    SGDK.BMP_clear()
    SGDK.BMP_setPixel(10, 10, 0xFF)
    SGDK.BMP_setPixel(20, 10, 0xFF)
    SGDK.BMP_setPixel(30, 10, 0xFF)

    SGDK.BMP_drawLine(line001)

    line001.x += 2
    line001.y += 1
    line001.w += 2
    line001.z += 1

    SGDK.BMP_flip(1);
    SGDK.BMP_waitFlipComplete()
    SGDK.c('}')
    SGDK.c('}')
    SGDK.end()  # stop writing main.c file


# ---------------------------ENTRY
#   GLOBALS
VMD = VMD()
SGDK = SGDK(VMD)
TRUE = SGDK.TRUE
FALSE = SGDK.FALSE
line001 = Vec4(50, 50, 100, 100)
main()

File: plot/python/2D_BitmapMode_proto/main.c
#include <genesis.h>
int main(){
VDP_setScreenWidth256();
BMP_init(TRUE,0,FALSE);
u16 vblCount = 0;
u16 circleSize = 100;
void circle(u16 x, u16 y, u16 size, u16 plotRes){
if(size == 0) size = 1;
u16 i = 0; for(i = 0; i < 1024; i += plotRes){
BMP_setPixel(x + (sinFix32(i) / size), (y + cosFix32(i) / size), 0xFF);
}
}
while(1){
BMP_clear();
BMP_showFPS(1);
circle(100, 100, circleSize, 30);
circle(100, 100, circleSize - 40, 30);
circle(100, 100, circleSize - 60, 30);
circleSize--;
vblCount++;
BMP_flip(1);
BMP_waitFlipComplete();
}
}
